config/condition_schedule.go

```diff
@@ -0,0 +1,102 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/cronexpr"
+)
+
+const scheduleConditionType = "schedule"
+
+var _ ConditionConfig = (*ScheduleConditionConfig)(nil)
+
+// ScheduleConditionConfig configures a condition configuration block of type
+// 'schedule'. A schedule condition is triggered by a configured cron schedule
+type ScheduleConditionConfig struct {
+	Cron *string `mapstructure:"cron"`
+}
+
+// Copy returns a deep copy of this configuration.
+func (c *ScheduleConditionConfig) Copy() ConditionConfig {
+	if c == nil {
+		return nil
+	}
+
+	var o ScheduleConditionConfig
+	o.Cron = StringCopy(c.Cron)
+
+	return &o
+}
+
+// Merge combines all values in this configuration with the values in the other
+// configuration, with values in the other configuration taking precedence.
+func (c *ScheduleConditionConfig) Merge(o ConditionConfig) ConditionConfig {
+	if c == nil {
+		if isConditionNil(o) { // o is interface, use isConditionNil()
+			return nil
+		}
+		return o.Copy()
+	}
+
+	if isConditionNil(o) {
+		return c.Copy()
+	}
+
+	r := c.Copy()
+	o2, ok := o.(*ScheduleConditionConfig)
+	if !ok {
+		return r
+	}
+
+	r2 := r.(*ScheduleConditionConfig)
+
+	if o2.Cron != nil {
+		r2.Cron = StringCopy(o2.Cron)
+	}
+
+	return r2
+}
+
+// Finalize ensures there no nil pointers.
+func (c *ScheduleConditionConfig) Finalize(services []string) {
```

Comment:
maybe since services isn't actually used, and since we're just adding it to satisfy the interface, we can make this anonymous?

`func (c *ScheduleConditionConfig) Finalize([]string)`